keyboards/togkey/pad_plus/pad_plus.c

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2025 TogKey */
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include "quantum.h"
+
+#if defined(OLED_ENABLE)
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    return OLED_ROTATION_180;
+}
+
+enum custom_keycodes { LAYER_SCROLL = QK_KB_0 };
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case LAYER_SCROLL:
+            if (!record->event.pressed) {
+                // Already handled the keycode, no further processing is needed
+                return false;
+            }
+
+            uint8_t current_layer = get_highest_layer(layer_state);
+            if (current_layer > 3 || current_layer < 0) {
+                return false;
+            }
+
+            uint8_t next_layer = current_layer + 1;
+            if (next_layer > 3) {
+                next_layer = 0;
+            }
+
+            layer_move(next_layer);
+
+            return false;
+        default:
+            return true; // Process all other keycodes normally
+    }
+}
+
+bool oled_task_user() {
+    static const char image_layer_1[] PROGMEM = {
```

Comment:
Would be preferable for these arrays to be moved outside of the `oled_task_kb()` function.